WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticListenerSubscriber.cs

```diff
@@ -2,14 +2,24 @@
 {
     using System;
     using System.Diagnostics;
+    using System.Diagnostics.Tracing;
     using Microsoft.ApplicationInsights.Extensibility;
 
-    internal sealed class AzureSdkDiagnosticListenerSubscriber : DiagnosticSourceListenerBase<object>
+    internal sealed class AzureSdkDiagnosticListenerSubscriber : DiagnosticSourceListenerBase<object>, IDisposable
     {
         public const string DiagnosticListenerName = "Azure.";
+        private readonly IDisposable logsListener;
 
         public AzureSdkDiagnosticListenerSubscriber(TelemetryConfiguration configuration) : base(configuration)
         {
+            // listen to Cosmos EventSource only - other logs can be sent using ILogger
```

Comment:
I understand this is draft PR, but can we just ask customers to leverage existing EventSource adapter to minimize changes here?
 
https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-trace-logs#use-eventsource-events

Reply:
Good point! I'd love to, but it'd mean users would need to install and configure `Microsoft.ApplicationInsights.EventSourceListener`, while we believe the events coming through this source are essential for users to debug any cosmos-related issues. 
Adding a dependency to `Microsoft.ApplicationInsights.EventSourceListener` from DependencyCollector would be an overkill.